api/environments/job_handler.go

```diff
@@ -168,15 +169,19 @@ func (eh EnvironmentHandler) getScheduledJobSummary(job *batchv1.Job,
 		Name:      job.Name,
 		Created:   radixutils.FormatTimestamp(creationTimestamp.Time),
 		Started:   radixutils.FormatTime(job.Status.StartTime),
-		Ended:     radixutils.FormatTime(job.Status.CompletionTime),
 		BatchName: batchName,
 	}
 
 	if jobPods, ok := jobPodsMap[job.Name]; ok {
 		summary.ReplicaList = getReplicaSummariesForPods(jobPods)
 	}
 	jobStatus := jobSchedulerApi.GetJobStatusFromJob(eh.kubeUtil.KubeClient(), job, jobPodsMap[job.Name])
-	summary.Status = jobStatus.Status
+	if jobPods, ok := jobPodsMap[job.Name]; ok && jobStatus.Status == jobModels.Failed.String() && jobHasExceededDeadline(jobPods) {
+		summary.Status = "DeadlineExceeded"
```

Comment:
I think all this can be done in the method `GetJobStatusFromJob`, especially created for this purpose? It has already selected pod for the job passed as an argument.